src/generate_price_reconciliation_report.py

```diff
@@ -0,0 +1,264 @@
+from configs.constants import TableNames
+import os
+import polars as pl
+import sqlite3
+from utils.utils import Utils
+
+
+def get_config_n_constant() -> tuple[str, str, str, str, str, str, str]:
+    """
+    Retrieves:
+        - Database path and output data path from config
+        - Table name from constants:
+            - FUND_HOLDINGS
+            - BOND_PRICES
+            - BOND_REFERENCE
+            - EQUITY_PRICES
+            - EQUITY_REFERENCE
+
+    Returns:
+        tuple[str, str, str, str, str, str, str]: A tuple containing the database path, output data path, and table names.
+    """
+    config = Utils.load_config()
+    database_path = config.database_path
+    output_data_path = config.output_data_path
+    fund_holdings_table_name = TableNames.FUND_HOLDINGS
+    bond_prices_table_name = TableNames.BOND_PRICES
+    bond_reference_table_name = TableNames.BOND_REFERENCE
+    equity_prices_table_name = TableNames.EQUITY_PRICES
+    return (
+        database_path,
+        output_data_path,
+        fund_holdings_table_name,
+        bond_prices_table_name,
+        bond_reference_table_name,
+        equity_prices_table_name,
+    )
+
+
+def get_bond_prices(
+    conn: sqlite3.Connection,
+    bond_prices_table_name: str,
+    bond_reference_table_name: str,
+) -> pl.DataFrame:
+    """
+    Retrieves bond prices from the database.
+    Joined with the bond reference table to get SEDOL as not all equities provided has ISIN.
+
+    Args:
+        conn (sqlite3.Connection): SQLite connection object.
+        bond_prices_table_name (str): Name of the bond prices table.
+        bond_reference_table_name (str): Name of the bond reference table.
+
+    Returns:
+        pl.DataFrame: DataFrame containing bond prices.
+    """
+    query = f"""SELECT 
+                br.SEDOL, 
+                bp.* 
+                FROM {bond_prices_table_name} bp
+                JOIN {bond_reference_table_name} br
+                ON bp.ISIN = br.ISIN"""
+    bond_prices_df = pl.read_database(query, conn)
+    return bond_prices_df
+
+
+def perform_price_reconciliation(
+    fund_holdings_df: pl.DataFrame,
+    bond_prices_df: pl.DataFrame,
+    equity_prices_df: pl.DataFrame,
+) -> pl.DataFrame:
+    """
+    Performs price reconciliation between fund holdings and bond/equity prices.
+
+    Args:
+        fund_holdings_df (pl.DataFrame): DataFrame containing fund holdings.
+        bond_prices_df (pl.DataFrame): DataFrame containing bond prices.
+        equity_prices_df (pl.DataFrame): DataFrame containing equity prices.
+
+    Returns:
+        pl.DataFrame: DataFrame containing the price reconciliation report.
+    """
+    # Clean and prepare the data
+    holdings_government_bonds_df = fund_holdings_df.filter(
```

Comment:
A long function like this may well be worth splitting into logical subcomponents. That would also help with testing.

This is translated SQL?

Performance is probably not an issue, but making so many new dataframes could be a problem eventually - you can use polars in lazy mode to be more efficient.